### PR TITLE
Plugins: consistent alpha behaviour across panel/datasource

### LINKED ISSUE #16529: Consistent alpha support across plugins
All plugin types should consistently support the `alpha` state.  They should:
 * only show up if `enable_alpha` is true
 * show a consistent warning (like the current datasource plugins)

I think:
 * alpha: only shows up if `enable_alpha` is true
 * beta: shows up for everyone, but adds the warning header

The setting is currently in:
```
[panels]
enable_alpha = false
```
Thats fine for now, but maybe we want to change that to [plugins]?

### PR DESCRIPTION
**Pull request by ryantxu**
fixes #16529

In #16528 I will look at adding this to App plugins

This will add a banner to the top of panels that looks like:
image

Text is moved from the datasource -- i did not write it :)

### CODE REVIEW

**Comment by torkelo** [comment]
nice!

File: pkg/api/frontendsettings.go [hunk 1]
```diff
@@ -162,6 +162,7 @@ func (hs *HTTPServer) getFrontendSettingsMap(c *m.ReqContext) (map[string]interf
 			"hideFromList": panel.HideFromList,
 			"sort":         getPanelSort(panel.Id),
 			"dataFormats":  panel.DataFormats,
+			"state":        panel.State,
```
Comment (ryantxu) [inline pkg/api/frontendsettings.go]: @torkelo -- now most panel plugins get:
```
 'state': '',
```
is there a good way in go to only add stat if it exists?

**Comment by ryantxu** [comment]
Trying to make a review on #16528 easier, this includes a few typescript changes that are not really necessary here

File: pkg/api/frontendsettings.go (hunk 1 above)
Comment (torkelo) [reply pkg/api/frontendsettings.go]: need to make dto for this and then we can add json attribute to not include the prop if empty. But leaving that as is for now.

**Comment by torkelo** [comment]
Looks good, made some refactorings, moved to [plugins] section. And updated docs. Added a bit of backward compatible logic to support old location for setting as well.

**Pull request merged**